sdk/core/azure-core/azure/core/pipeline/policies/_base.py

```diff
@@ -149,7 +149,10 @@ class RequestHistory(object):
 
     def __init__(self, http_request, http_response=None, error=None, context=None):
         # type: (PipelineRequest, Optional[PipelineResponse], Exception, Optional[Dict[str, Any]]) -> None
-        self.http_request = copy.deepcopy(http_request)
+        try:
+            self.http_request = copy.deepcopy(http_request)
+        except ValueError:
```

Comment:
under what circumstance can this be triggered?

Reply:
Per the bug:
If the request was created with a stream as input, the deepcopy will fail:

https://github.com/Azure/azure-sdk-for-python/issues/7732

Reply:
Fair enough, I didn't see the bug referenced. So then I'd ask:
* if shallow copies work sometimes, can they just be used all the time? 
* if not, what about  implementing `__deepcopy__` on `PipelineRequest` and handling this situation there, so that `deepcopy` can be used here unconditionally?

Ownership is hard enough to manage when there is consistent behavior, I'd be worried about subtle effects of sometimes having deep copies and sometimes not.

Reply:
http_request is not a pipeline request object. Do you mean transport.HttpRequest?

Reply:
I was going off what the type annotation says:
```
 def __init__(self, http_request, http_response=None, error=None, context=None):
        # type: (PipelineRequest, Optional[PipelineResponse], Exception, Optional[Dict[str, Any]]) -> None
```
Regardless, my suggestion is to try to avoid different behaviors at different times here.

Reply:
I agree with @bryevdv , let's do a `__deepcopy__` in the request.